connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -1073,6 +1074,12 @@ class SparkConnectPlanner(val session: SparkSession) {
         }
         Some(Lead(children.head, children(1), children(2), ignoreNulls))
 
+      case "bloom_filter_agg" if fun.getArgumentsCount == 3 =>
+        val children = fun.getArgumentsList.asScala.toSeq.map(transformExpression)
+        Some(
+          new BloomFilterAggregate(children.head, children(1), children(2))
```

Comment:
You will need to hash the input column.

Reply:
There is a small issue here. The aggregate requires the first input to be a Long. `DataFrameStatFunctions.bloomFilter` supports `Byte`, `Short`, `Int`, `Long`, and `String`. While we can simply add a cast to long for the first 4, string will be an issue. We need adapt the BloomFilterAggregate to make it fully compatible.

Reply:
> You will need to hash the input column.

like `new BloomFilterAggregate(new XxHash64(Seq(children.head)), ....`

But if we hash the input column, I think it will be hashed twice ... the existing test case 

```
assert(0.until(1000).forall(filter1.mightContain))
```

will not pass

Reply:
> There is a small issue here. The aggregate requires the first input to be a Long. `DataFrameStatFunctions.bloomFilter` supports `Byte`, `Short`, `Int`, `Long`, and `String`. While we can simply add a cast to long for the first 4, string will be an issue. We need adapt the BloomFilterAggregate to make it fully compatible.

So maybe adding a new protobuf message is a more simpler way? `BloomFilterAggregate` is a internal function used by `InjectRuntimeFilter`,  and `InjectRuntimeFilter` hashes both input column and might contain values, so I think there is no String type input in that scenario?

Reply:
Refactored `BloomFilterAggregate ` to support StringType